common/tqid/task_queue_id.go

```diff
@@ -31,31 +31,39 @@ import (
 	"strings"
 
 	enumspb "go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/serviceerror"
 	taskqueuepb "go.temporal.io/api/taskqueue/v1"
 	"go.temporal.io/server/common/namespace"
 )
 
 const (
-	// NonRootPartitionPrefix is the prefix for all mangled task queue names.
-	NonRootPartitionPrefix = "/_sys/"
-	PartitionDelimiter     = "/"
+	// nonRootPartitionPrefix is the prefix for all mangled task queue names.
+	nonRootPartitionPrefix = "/_sys/"
+	partitionDelimiter     = "/"
 )
 
 type (
-	// TaskQueue represents the high-level task queue that user create by explicitly providing a TaskQueue name
-	// when starting a worker or a workflow. Under the hood, a TaskQueue can be broken down to multiple sticky
-	// or normal partitions.
-	TaskQueue struct {
+	// TaskQueueFamily represents the high-level "task queue" that user creates by explicitly providing a task queue name
+	// when starting a worker or a workflow. A task queue family consists of separate TaskQueue's for different types of
+	// task.
+	TaskQueueFamily struct {
 		namespaceId namespace.ID
-		// this can be string as long as it does not start with /_sys/.
+		// this can be any string as long as it does not start with /_sys/.
 		name string
 	}
 
+	// TaskQueue represents a logical task queue for a type of tasks (e.g. Activity or Workflow). Under the hood,
+	// a TaskQueue can be broken down to multiple sticky or normal partitions.
+	TaskQueue struct {
+		family   TaskQueueFamily
```

Comment:
I did consider that, but decided not to do it because a task queue is not a family. some methods in family do not make sense to be added to TaskQueue.